presto-main/src/main/java/com/facebook/presto/operator/aggregation/histogram/HistogramStateFactory.java

```diff
@@ -39,21 +42,41 @@ public HistogramStateFactory(
         this.mode = mode;
     }
 
+    public HistogramStateFactory(
```

Comment:
instead of extending the arguments, I think you should keep the `NdvEstimatorStateFactory` class, but then just make that a subclass of the `HistogramStateFactory`. I think adding this additional `ndvEstimate` field is a little confusing here as it doesn't have anything to do with the histogram state. Plus, it's cleaner to keep all of the logic contained to the `estimatendv` package